[EWL-5018] SG2 | Code standards to-dos for Resource page

JaeBea opened the pull request:
## Ticket(s)

**Jira Ticket**
- [EWL-5018: SG2 | Code standards to-dos for Resource page](https://issues.ama-assn.org/browse/EWL-5018)

## Description
As a developer using the Resource page pattern in SG2, the pattern should be built consistently with Twig and utilize the jQuery UI Tabs library.

This update refactors relevant twig templates, json, js, and scss for resource page tab functionality.

## To Test

- Pull `feature/EWL-5018-code-standards-resource-page` branch
- Run `gulp serve` in ama-style-guide-2/styleguide to ensure a local version of style guide is running.
- Navigate to [Page > Resource](http://localhost:3000/?p=pages-resource).
- Confirm page still works as it did in [Style Guide 2](https://americanmedicalassociation.github.io/ama-style-guide-2/?p=pages-resource)


## Visual Regressions

Updated reference images for resource page backstop tests have been included in this PR.

## Relevant Screenshots/GIFs
N/A


## Remaining Tasks
N/A


## Additional Notes
N/A

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

JaeBea committed:
Switching resource navigation functionality to JqueryUI

JaeBea committed:
Fixing display in IE and updating styles for use with jQuiry UI.

JaeBea committed:
Merge branch 'develop' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into feature/EWL-5018-code-standards-resource-page

# Conflicts:
#	styleguide/source/_patterns/03-organisms/resource-tabs.twig

JaeBea committed:
Add tabs json structure.

JaeBea committed:
Merge branch 'develop' into feature/EWL-5018-code-standards-resource-page

JaeBea committed:
Updating twig template to use existing atoms and molecules.

JaeBea committed:
Merge branch 'develop' into feature/EWL-5018-code-standards-resource-page

JaeBea committed:
Update link icon classes, link icon styles and ama__tabs related scripts.

JaeBea committed:
Update styles and templates to remove regressions.

JaeBea committed:
Updated backstop references for resource related pages and organisms.

designerneil commented:
The related links tab missing the subtitle under the CME link
<img width="616" alt="screen shot 2018-07-10 at 11 05 20 am" src="https://user-images.githubusercontent.com/2271747/42522662-5099bd18-8431-11e8-8d97-c113e5c5cca3.png">
<img width="588" alt="screen shot 2018-07-10 at 11 05 13 am" src="https://user-images.githubusercontent.com/2271747/42522664-50ad46d0-8431-11e8-8ddd-514802e82d05.png">

designerneil commented on styleguide/source/_patterns/05-pages/resource.twig:
```diff
@@ -4,7 +4,7 @@
 {% include '@organisms/masthead/masthead.twig' %}
 <div class="ama____page--resources__page-content">
   <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad asperiores atque cupiditate <a href="http://www.google.com">normal link</a> debitis dignissimos, dolorem dolorum ea
-  <a href="#" class="ama__page--resource__resource-link" data-resource="downloads-1">(Download 1)</a> eius hic laboriosam, necessitatibus nesciunt nobis omnis placeat reprehenderit tenetur ullam vel veniam. Delectus, eaque est molestiae pariatur quae quidem suscipit tenetur voluptatibus.</p>
```
Can we use the link atom here?

designerneil commented on styleguide/source/assets/js/tabs.js:
```diff
@@ -1,57 +1,27 @@
+    if (viewportWidth >= 600 && $('.ama__resource-tabs').length > 0) {
+        defaultActiveTab = 1;
+    }
+    $(".ama__tabs").tabs({
```
Well done!

designerneil commented on styleguide/source/assets/js/tabs.js:
```diff
@@ -1,57 +1,27 @@
+    });
+
+    // Prevent jump onclick
+    $('.ama__resource-tabs__nav a, .ama__tabs-navigation a').on('click', function(e){
```
Page seems to still jump. I tested it by making my page a bit shorter and clicking on the tabs

designerneil commented:
<img width="1439" alt="screen shot 2018-07-10 at 11 33 44 am" src="https://user-images.githubusercontent.com/2271747/42524110-26b3daca-8435-11e8-854e-8fce24abb635.png">

designerneil commented:
The resource.twig has hard coded text that probably should go into the json file:

```
<div class="ama____page--resources__page-content">
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad asperiores atque cupiditate <a href="http://www.google.com">normal link</a> debitis dignissimos, dolorem dolorum ea
  <a href="#downloads" class="ama__page--resource__resource-link" >(Download 1)</a> eius hic laboriosam, necessitatibus nesciunt nobis omnis placeat reprehenderit tenetur ullam vel veniam. Delectus, eaque est molestiae pariatur quae quidem suscipit tenetur voluptatibus.</p>
  {% include '@molecules/resource-link-as-inline.twig' with { resourceLinkAsInline : body.resourceLinkAsInline } %}
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aperiam assumenda consequuntur cupiditate, ea eius, facilis fuga laudantium modi molestiae placeat similique soluta tempora vel? Aliquid cupiditate delectus distinctio, doloribus enim eveniet illum, laudantium obcaecati pariatur perspiciatis quaerat quod similique velit. Eos libero provident quod sit.</p>
</div>
```

designerneil commented:
Resouce.md need the json structure in it:

```
### Description
A page using the Split layout that houses resources based on the page content.

[EWL-4921](https://issues.ama-assn.org/browse/EWL-4921)

```

designerneil commented:
The columns seem thinner than in gh-pages causing the header to wrap differently (probably being cause by either a margin or padding around the container, ):

**localhost:**
<img width="1231" alt="screen shot 2018-07-10 at 11 40 35 am" src="https://user-images.githubusercontent.com/2271747/42524461-29f7f440-8436-11e8-9ed1-43a41e2c8891.png">


**gh-pages:**
<img width="1205" alt="screen shot 2018-07-10 at 11 41 33 am" src="https://user-images.githubusercontent.com/2271747/42524479-381285c2-8436-11e8-8098-db67c1291392.png">

JaeBea committed:
Correcting syntax and removing hardcoded paragraph text.

JaeBea committed:
Updating backstop images.

JaeBea committed:
Merge branch 'develop' into feature/EWL-5018-code-standards-resource-page

JaeBea commented:
@designerneil I have addressed all concerns except the jump when clicking a tab. I do not know how to prevent this without removing the hash completely, which causes issues when javascript is not enables. What should work, e.preventDefault(), is not consistently working. I feel we should table that issue and address at a later time as other committed areas using jQueryUI tabs have the same issue.
